spec.md

```diff
@@ -379,6 +390,17 @@ This string MAY be surfaced by CO to end users.
 
 The status `details` MUST be empty. In the future, this spec may require `details` to return a machine-parsable protobuf message if the status `code` is not `OK` to enable CO's to implement smarter error handling and fault resolution.
 
+### Secrets Requirements
+
+Secrets may be required by plugin to complete a RPC request.
+A secret is a string to string map where the key identifies the name of the secret (e.g. "username" or "password"), and the value contains the secret data (e.g. "bob" or "abc123").
+Each key MUST consist of alphanumeric characters, '-', '_' or '.'.
+Each value MUST contain a valid string. An SP MAY choose to accept binary (non-string) data by using a binary-to-text encoding scheme, like base64.
+An SP SHALL advertise the requirements for required secret keys and values in documentation.
+CO SHALL permit passing through the required secrets.
+A CO MAY pass the same secrets to all RPCs, therefore the keys for all unique secrets that an SP expects must be unique across all CSI operations.
+This information is sensitive and MUST be treated as such (not logged, etc.) by the CO.
+
```

Comment:
nit: I prefer having this information on each field instead of in a common place because it makes it easy for a reader to discover the requirements when it is a comment on the field rather then an independent paragraph that they have to find.

Reply:
I moved this info to a common place to address comments from @julian-hj.  He said the comments are getting very lengthy for each field and it's better to put the comments in a common place.  I'm fine doing it either way.

Reply:
I actually like having them here, honestly.

Reply:
@saad-ali would it help if the secrets field in the protos referred back to this section for requirements?

Reply:
@jdef That's a fair compromise. I would like the proto file to be self-documenting. If there is a requirement for some field, I want that to be quickly apparent from reading the comment rather then require user to go read the entire document. If we can have a pointer comment saying "see section x of spec for requirements on y", that should be fine -- since the reader will be aware that there are additional requirements that apply.